### PR TITLE
Feature/gh 5101

### LINKED ISSUE #5101: Add "get the details of returns request" in OCC, store and facade layer
Given `returnRequestCode`, we can get the details of the return request. We need add it into these layers:

- [ ] OCC
- [ ] store
- [ ] facades

### PR DESCRIPTION
**Pull request by WeizhengSap**
close #5101

### CODE REVIEW

File: projects/core/src/user/store/reducers/order-return-request.reducer.ts [hunk 1]
```diff
@@ -1,20 +1,17 @@
...
+): ReturnRequest {
   switch (action.type) {
+    case UserActions.LOAD_ORDER_RETURN_REQUEST_SUCCESS:
     case UserActions.CREATE_ORDER_RETURN_REQUEST_SUCCESS: {
       const returnRequest: ReturnRequest = action.payload;
-
-      return {
-        returnRequest,
-      };
+      return returnRequest;
```
Comment (plabadie) [inline projects/core/src/user/store/reducers/order-return-request.reducer.ts]: Perhaps simply: `return action.payload;`

File: projects/core/src/user/store/reducers/order-return-request.reducer.ts [hunk 2]
```diff
@@ -1,20 +1,17 @@
...
-export const returnRequestInitialState: OrderReturnRequestState = {};
+export const returnRequestInitialState = {};
 
 export function returnRequestReducer(
   state = returnRequestInitialState,
   action: UserActions.OrderReturnRequestAction
-): OrderReturnRequestState {
+): ReturnRequest {
   switch (action.type) {
+    case UserActions.LOAD_ORDER_RETURN_REQUEST_SUCCESS:
```
Comment (plabadie) [inline projects/core/src/user/store/reducers/order-return-request.reducer.ts]: I'm probably missing something, but I don't think handling `LOAD_ORDER_RETURN_REQUEST_SUCCESS` in a custom reducer is required.  It is a `LoaderSuccessAction` and the meta reducer will already put the action payload and sign it to the loader state value. 


In fact, the same goes for `CREATE_ORDER_RETURN_REQUEST_SUCCES`.

File: projects/core/src/user/facade/order-return-request.service.ts [hunk 3]
```diff
@@ -12,6 +12,7 @@ import { StateWithProcess } from '../../process/store/process-state';
 import { UserActions } from '../store/actions/index';
 import { UsersSelectors } from '../store/selectors/index';
 import { StateWithUser } from '../store/user-state';
+import { OCC_USER_ID_CURRENT } from '../../occ/index';
 
 @Injectable({
   providedIn: 'root',
```
[the remark this answers is not in the document]
Comment (WeizhengSap) [reply projects/core/src/user/facade/order-return-request.service.ts]: I don't know. I just feel `user-order-return-request` is a long name :).

File: projects/core/src/user/store/actions/order-return-request.action.ts [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
 import {
   ReturnRequest,
   ReturnRequestEntryInputList,
   ReturnRequestList,
 } from '../../../model/order.model';
-import { USER_RETURN_REQUESTS } from '../user-state';
+import {
+  USER_RETURN_REQUESTS,
+  USER_RETURN_REQUEST_DETAILS,
+} from '../user-state';
```
[the remark this answers is not in the document]
Comment (WeizhengSap) [reply projects/core/src/user/store/actions/order-return-request.action.ts]: `user-state` is a `ts` file, it is not a folder.

File: projects/core/src/user/store/reducers/order-return-request.reducer.ts (hunk 2 above)
Comment (WeizhengSap) [reply projects/core/src/user/store/reducers/order-return-request.reducer.ts]: yes, you are right. I will remove this reducer.

File: projects/core/src/user/connectors/order/user-order.connector.ts [hunk 5]
```diff
@@ -52,6 +52,13 @@ export class UserOrderConnector {
     return this.adapter.createReturnRequest(userId, returnRequestInput);
   }
 
+  public getReturnRequestDetail(
```
[the remark this answers is not in the document]
Comment (WeizhengSap) [reply projects/core/src/user/connectors/order/user-order.connector.ts]: I don't know. function `get` calls `adapter.load`, function `getHistory` calls `adatper.loadHistory`. In this `connector`, there is no `load*`.

File: projects/core/src/user/store/effects/order-return-request.effect.ts [hunk 6]
```diff
@@ -35,11 +34,36 @@ export class OrderReturnRequestEffect {
     })
   );
 
+  @Effect()
+  loadReturnRequest$: Observable<
+    UserActions.OrderReturnRequestAction
+  > = this.actions$.pipe(
+    ofType(UserActions.LOAD_ORDER_RETURN_REQUEST),
+    map((action: UserActions.LoadOrderReturnRequest) => action.payload),
+    switchMap(payload => {
```
[the remark this answers is not in the document]
Comment (WeizhengSap) [reply projects/core/src/user/store/effects/order-return-request.effect.ts]: we want to cancel concurrent requests :).

File: projects/core/src/user/facade/order-return-request.service.ts [hunk 7]
```diff
@@ -30,25 +31,31 @@ export class OrderReturnRequestService {
...
+      select(UsersSelectors.getOrderReturnRequestState),
+      tap(returnState => {
+        const attemptedLoad =
+          returnState.loading || returnState.success || returnState.error;
+        if (!attemptedLoad) {
+          this.loadOrderReturnRequestDetail(returnRequestCode);
+        }
+      }),
+      map(returnState => returnState.value)
+    );
```
[the remark this answers is not in the document]
Comment (WeizhengSap) [reply projects/core/src/user/facade/order-return-request.service.ts]: I will add 'return request code' check.

File: projects/core/src/user/store/actions/order-return-request.action.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (plabadie) [reply projects/core/src/user/store/actions/order-return-request.action.ts]: Thanks. My bad!

**Pull request merged**